### PR TITLE
10 a suitable lesson section

### PR DESCRIPTION
**Pull request by NLMOTS**
I checked the functionality, according to the list of requirements, everything seems to work

### CODE REVIEW

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.html [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                <button class="button-icon" mat-icon-button [matMenuTriggerFor]="appMenu">
+                    <mat-icon>calendar_today</mat-icon>
+                </button>
+                <mat-menu #appMenu="matMenu">
+                    <mat-calendar #calendar (selectedChange)="select($event, calendar)" [dateClass]="isSelected">
+                    </mat-calendar>
+                </mat-menu>
+                {{ getDayNumber(days[0].getDate()) }} {{ getMonthString(days[0]) }} -
+                {{ getDayNumber(days[6].getDate()) }}
+                {{ getMonthString(days[6]) }}
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.html:19]: Can you make this in one function or use date formatting directive

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.html [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        </div>
+        <div class="days">
+            <div
+                class="day-elem"
+                [ngClass]="{
+                    'day-elem-blue': item.getDate() === todayDate.getDate() && item.getMonth() === todayDate.getMonth(),
+                    'day-elem-grey': compareDate(item, todayDate)
+                }"
+                *ngFor="let item of days">
+                {{ getDayNumber(item.getDate()) }} {{ getMonthString(item) }}
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.html:35]: The same here

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.html [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                <h2>{{ getWeekDayString(item) }}</h2>
+                <div class="labels">
+                    <div
+                        *ngIf="item.getDate() === todayDate.getDate() && item.getMonth() === todayDate.getMonth()"
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.html:39]: In one function

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.html [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                            'mettings-grey': compareDate(item, todayDate)
+                        }">
+                        today
+                    </div>
+                    <div
+                        class="mettings"
+                        [ngClass]="{
+                            'mettings-grey': compareDate(item, todayDate)
+                        }">
+                        6 meetings
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.html:51]: Make it loaded from ts

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.sass [hunk 5]
```diff
@@ -0,0 +1,126 @@
+@import '../../../../styles/colors'
+@font-face 
+  font-family: "mainfont"
+  src: url("../../../../assets/fonts/Barlow-Medium.ttf") format("truetype")
+
+$inactiveDayFont: rgb(181,183,196)
+$inactiveDayBackground: rgb(224, 226, 242)
+$inactiveDayLabel: rgb(201,203,218)
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.sass:8]: If we don't have this colors in styles, please add them globally

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.sass [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    align-items: center
+    background-color: $white
+    padding: 25px
+    font-family: mainfont
+
+.container
+    display: flex
+    flex-direction: column
+    align-items: center
+    background-color: rgb(246, 245, 250)
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.sass:21]: All colors to styles/colors

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.sass [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+    h2
+        margin: 0
+        margin-bottom: 10px
+        margin-top: 5px
+    
+.day-elem-blue
+    background-color: $main-color
+    color: $white
+.day-elem-grey
+    background-color: $inactiveDayBackground
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.sass:79]: Add new lines between different class styles

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.sass [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    border-radius: 15px
+    text-align: center
+    font-size: 12px
+    margin-right: 5px
+    height: 24px
+    width: 77px
+.mettings-grey
+    background-color: $inactiveDayLabel
+    color: $inactive-color
+.ChooseDay
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.sass:97]: Name all classes in one format small letters, hyphen between

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts [hunk 9]
```diff
@@ -0,0 +1,111 @@
+import { Component } from '@angular/core';
+
+@Component({
+    selector: 'app-suitable-lesson',
+    templateUrl: './suitable-lesson.component.html',
+    styleUrls: ['./suitable-lesson.component.sass'],
+})
+export class SuitableLessonComponent {
+    update: any = true;
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts:9]: Check github actions

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+        'April',
+        'May',
+        'June',
+        'July   ',
+        'August',
+        'September',
+        'October',
+        'November',
+        'December',
+    ];
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts:26]: You can get it from angular date directive

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+    dayOfWeek: any;
+
+    startDate: Date = new Date();
+
+    currentDate: Date = new Date();
+
+    selectedDate: Date = new Date();
+
+    isSelected: any;
+
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts:41]: Are you sure we need all this variables?

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+    }
+
+    getDayNumber(inNumber: number): string {
+        if (inNumber > 9) {
+            return inNumber.toString();
+        }
+
+        // eslint-disable-next-line prefer-template
+        return '0' + inNumber.toString();
+    }
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts:82]: Date directive

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.sass (hunk 6 above)
Comment (Anatolii-Ivanov) [reply frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.sass:17]: Still not fixed

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+    selector: 'app-suitable-lesson',
+    templateUrl: './suitable-lesson.component.html',
+    styleUrls: ['./suitable-lesson.component.sass'],
+})
+export class SuitableLessonComponent {
+    days: Date[] = [];
+
+    selectedDate: Date = new Date();
+
+    isSelected: any;
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts:14]: Github actions...

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+        const tempDate: Date = this.selectedDate;
+
+        const tempDayOfWeek: number = this.selectedDate.getDay();
+
+        for (let i = 0; i < 7; i++) {
+            const day: Date = new Date(
+                tempDate.getFullYear(),
+                tempDate.getMonth(),
+                tempDate.getDate() + i - tempDayOfWeek + 1,
+            );
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts:74]: Maybe also moment and add days to first date

**Comment by Anatolii-Ivanov** [comment]
Please, resolve fixed comments

**Review by Anatolii-Ivanov — DISMISSED** [review]
Good job :)

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    weekDec(): void {
+        this.selectedDate.setDate(this.selectedDate.getDate() - 7);
+
+        this.setDays();
+    }
+
+    getWeekDayString(inDate: Date): string {
+        return new DatePipe('en-US').transform(inDate, 'EEEE') ?? '';
```
Comment (tatianahutii) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts:51]: Please do not use magic numbers

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+    getMonthString(inDate: Date): string {
+        return new DatePipe('en-US').transform(inDate, 'MMMM') ?? ' ';
+    }
+
+    getDayNumber(inNumber: number): string {
+        if (inNumber > 9) {
+            return inNumber.toString();
+        }
+
+        // eslint-disable-next-line prefer-template
```
Comment (tatianahutii) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts:63]: see the previous comments

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+        return '0' + inNumber.toString();
+    }
+
+    getCalendarMonthString(): string {
+        const dateFormat = 'dd MMMM';
+        const formattedDate1 = new DatePipe('en-US').transform(this.days[0], dateFormat);
+        const formattedDate2 = new DatePipe('en-US').transform(this.days[6], dateFormat);
+
+        return `${formattedDate1} - ${formattedDate2}`;
```
Comment (tatianahutii) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts:72]: please refactor these lines

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts [hunk 18]
```diff
@@ -0,0 +1,88 @@
...
+        const tempDayOfWeek: number = this.selectedDate.getDay();
+
+        for (let i = 0; i < 7; i++) {
+            const day = moment(tempDate)
+                .add(i - tempDayOfWeek + 1, 'day')
+                .toDate();
+
+            this.days[i] = day;
+        }
+    }
```
Comment (tatianahutii) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts:87]: and refactor this as well

File: frontend/src/app/modules/time-table/time-table.module.ts [hunk 19]
```diff
@@ -0,0 +1,35 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (tatianahutii) [inline frontend/src/app/modules/time-table/time-table.module.ts:2]: see the previous comments

File: frontend/src/app/modules/time-table/time-table.module.ts [hunk 20]
```diff
@@ -0,0 +1,35 @@
...
+        BrowserModule,
+        CommonModule,
+        BrowserAnimationsModule,
+        MatFormFieldModule,
+        MatDatepickerModule,
+        MatInputModule,
+        MatNativeDateModule,
+        FormsModule,
+        MatMenuModule,
+        MatIconModule,
```
Comment (tatianahutii) [inline frontend/src/app/modules/time-table/time-table.module.ts:30]: This should be a part of Material Module

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/time-table/time-table.module.ts (hunk 20 above)
Comment (Anatolii-Ivanov) [reply frontend/src/app/modules/time-table/time-table.module.ts:30]: Still not fixed

File: frontend/src/app/modules/time-table/time-table.module.ts (hunk 19 above)
Comment (Anatolii-Ivanov) [reply frontend/src/app/modules/time-table/time-table.module.ts:2]: Still not fixed

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts (hunk 15 above)
Comment (Anatolii-Ivanov) [reply frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts]: Still not fixed
Move amount of days in week (1/7) to const variables at the top of the class

**Review by Anatolii-Ivanov — DISMISSED** [review]

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by SidletskyiOlexandr — APPROVED** [review]

**Pull request merged**